packages/roosterjs-editor-core/lib/coreApi/selectTable.ts

```diff
@@ -0,0 +1,206 @@
+import {
+    EditorCore,
+    SelectionRangeTypes,
+    TableSelection,
+    SelectTable,
+} from 'roosterjs-editor-types';
+
+const TABLE_ID = 'tableSelected';
+const CONTENT_DIV_ID = 'contentDiv_';
+
+let styleElement: HTMLStyleElement;
+
+/**
+ * @internal
+ * Select a table and save data of the selected range
+ * @param core The EditorCore object
+ * @param table table to select
+ * @param coordinates first and last cell of the selection, if this parameter is null, instead of
+ * selecting, will unselect the table.
+ * @returns true if successful
+ */
+export const selectTable: SelectTable = (
+    core: EditorCore,
+    table: HTMLTableElement,
+    coordinates?: TableSelection
+) => {
+    const doc = core.contentDiv.ownerDocument;
+
+    if (coordinates) {
+        ensureUniqueContentDivId(core);
+        table.id = ensureUniqueId(table, doc);
+        const ranges = select(table, coordinates, core);
+        return {
+            type: SelectionRangeTypes.TableSelection,
+            ranges,
+            table,
+            areAllCollapsed: ranges.filter(range => range?.collapsed).length == ranges.length,
+            coordinates,
+        };
+    }
+
+    unselect();
+
+    if (!coordinates && styleElement) {
+        styleElement.parentElement.removeChild(styleElement);
+        styleElement = null;
+    }
+    return null;
+};
+
+function buildCss(
+    table: HTMLTableElement,
+    coordinates: TableSelection,
+    contentDivSelector: string
+): { css: string; ranges: Range[] } {
+    coordinates = normalizeTableSelection(coordinates);
+    const tr1 = coordinates.firstCell.y + 1;
+    const td1 = coordinates.firstCell.x + 1;
+    const tr2 = coordinates.lastCell.y + 1;
+    const td2 = coordinates.lastCell.x + 1;
+    const ranges: Range[] = [];
+    const tbodySelector = table.querySelector('tbody') ? '>tbody' : '';
+    let firstSelected: HTMLTableCellElement | null = null;
+    let lastSelected: HTMLTableCellElement | null = null;
+
+    let css = '';
+
+    let isFirst = true;
+    for (let i = tr1; i <= tr2; i++) {
+        firstSelected = null;
+        lastSelected = null;
+
+        for (let j = td1; j <= td2; j++) {
+            if (isFirst) {
+                isFirst = false;
+            } else {
+                css += ',';
+            }
+
+            let selector =
+                contentDivSelector +
+                ' #' +
+                table.id +
+                tbodySelector +
+                ' >tr:nth-child(' +
+                i +
+                ')>td:nth-child(' +
+                j +
+                ')';
+            if (table.querySelector(selector)) {
+                css += selector;
+            } else {
+                selector = selector.replace('td', 'th');
+                if (table.querySelector(selector)) {
```

Comment:
Maybe we don't need this check, but just add whatever we have. Because if it fails the check, we still can do nothing